### PR TITLE
fix: Apply batched updates to resize observer callback

### PR DESCRIPTION
**Pull request by just-boris**
*Issue #, if available:*

*Description of changes:*

Optimization for React, if there will be multiple state changes inside `onObserve`, they will be batched into a single re-render

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by jperals** [comment]
Has this been somehow tested?

File: src/internal/container-queries/use-resize-observer.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
+import { unstable_batchedUpdates } from 'react-dom';
```
Comment (jperals) [inline src/internal/container-queries/use-resize-observer.ts:4]: Note that react-dom is a dev dependency —is this okay? Consumers of this package might not get it installed

Comment (just-boris) [reply src/internal/container-queries/use-resize-observer.ts:4]: We already use it in this package: https://github.com/search?q=repo%3Acloudscape-design%2Fcomponent-toolkit%20unstable_batchedUpdates&type=code

**Comment by just-boris** [comment]
> Has this been somehow tested?

Only manually. This API changes internal react things, you cannot write a test on this

**Review by jperals — APPROVED** [review]

**Pull request merged**